src/v/raft/consensus.cc

```diff
@@ -2247,6 +2247,7 @@ ss::future<> consensus::hydrate_snapshot() {
         co_await truncate_to_latest_snapshot(truncate_cfg.value());
     }
     _snapshot_size = co_await _snapshot_mgr.get_snapshot_size();
+    update_follower_stats(_configuration_manager.get_latest());
```

Comment:
I was wondering if we should even hydrate a snapshot if the included_index < start offset, just return success and make it idempotent?

Reply:
I looked at this again, I think the fix makes sense but I think the  issue though is L2241 already updates follower stats from snapshot config, so we are effectively doing it twice? Perhaps we should just swap  L2242 & L2241 and make update_offset_from_snapshot() use latest config?

Reply:
i've removed the stats update from `update_offset_from_snapshot`. I am not sure if the stats should be updated from within that method as its name suggest updating offsets.